src/catalog/src/expr_cache.rs

```diff
@@ -210,9 +199,7 @@ impl ExpressionCache {
                         };
                         // Remove dropped IDs and local expressions that were cached with different
                         // features.
-                        if !current_ids.contains(&key.id)
-                            || expressions.optimizer_features != *optimizer_features
-                        {
+                        if !current_ids.contains(&key.id) {
```

Comment:
why'd you move the check for optimizer features out to call-sites?

Reply:
It turns out that during startup, depending on the context, we fiddle around with the optimizer features. So there's no single `OptimizerFeatures` that can be used for all expressions.